virtio-devices/src/net.rs

```diff
@@ -90,7 +92,14 @@ impl NetEpollHandler {
             error!("Failed to get rx queue event: {:?}", e);
         }
 
-        if !self.net.rx_tap_listening {
+        let rate_limit_reached = self
+            .net
+            .rx_rate_limiter
+            .as_ref()
+            .map_or(false, |r| r.is_blocked());
+
```

Comment:
In here, you should set the new boolean that will store the information about some RX buffers being ready. Even if the limit is reached at the time the RX buffer is added, it's important to have this information as it will help decide if the TAP fd should be registered or not when the `RX_RATE_LIMITER_EVENT` is triggered.